lib/modules/userTagger.js

```diff
@@ -740,15 +740,25 @@ let currentSortMethod, isDescending;
 async function drawUserTagTable(sortMethod, descending) {
 	currentSortMethod = sortMethod || currentSortMethod;
 	isDescending = (descending === undefined || descending === null) ? isDescending : descending;
-	const filterType = $('#tagFilter').val();
 	const tagsData = await tagStorage.getAll();
-	const tags = filterMap(Object.entries(tagsData), ([k, v]) => {
+	const filterType = $('#tagFilter').val();
+	let tags = filterMap(Object.entries(tagsData), ([k, v]) => {
 		const tag = Tag.getUnfilled(k);
 		tag.load(v);
 		if (filterType === 'tagged users' && !tag.text) return;
 		return [tag];
 	});
 
+	// show 'all users' if there are no 'tagged users'
+	if (filterType === 'tagged users' && tags.length === 0) {
+		$('#tagFilter').val(i18n('userTaggerAllUsers'));
```

Comment:
In order to avoid loading all tags twice, a cleaner solution may be to build a complete `tags`, then creating the `#tagFilter` select with the default value depending on whether there's any tags
`<select id="tagFilter" value="${tags.find(v => v.text) ? 'tagged' : 'all'}">`, and don't do any filtering before then.

Reply:
When should `tags` be built? It currently looks like it's being rebuilt every single time the table needs to be drawn. So that's kind of excessive? 

When the `#tagFilter` select is created initially, it's not aware of the tags since those haven't been built yet. So should I build the tags once only when the dashboard functionality is being added (in `function addDashboardFunctionality` instead of `function drawUserTagTable`) and then load those same tags every time the table needs to be redrawn? (I'm guessing by passing them to `drawUserTagTable` as function args?)

Additionally should the tagged users option never be shown if there are no tagged users, or should it only show all users at first, initially by default, when there are no tagged users? The latter of which the user can select 'tagged users' but it will still only show an empty table.

Reply:
Passing it as a function arg sounds good, but since `tagStorage` may change, it may be easier to split `tagStorage` → `Array<Tag>` into a new function and call it from both places.

I think both options can stay regardless of there being any matching tags, as not to make this too complex.